deps-mvn-indexing/src/main/clojure/com/connexta/osgeyes/index/core.clj

```diff
@@ -67,4 +68,76 @@
   (count (gather-hierarchy "ddf" "2.19.5"))
   (map #(artifact-info->map % #{:attrs}) (do-gather-hierarchy "ddf" "ddf" "2.19.5"))
   (map #(.toString %) (do-gather-hierarchy "ddf" "ddf" "2.19.5"))
-  (close-indexer!))
\ No newline at end of file
+  (close-indexer!))
+
+(def test-data
+  '({:artifact-id "a"
+     :attrs       {"JAR_PACKAGES" "package_1,package_2"}}
+    {:artifact-id "b"
+     :attrs       {"JAR_PACKAGES" "package_2,package3"}})
+  )
+
+(comment
+  (open-indexer!)
+  (filter #(and (= "bundle" (:packaging %))
+                (.contains (:artifact-id %) "core")
+                (.contains (:artifact-id %) "security"))
+          (gather-hierarchy "ddf" "2.19.5"))
+  (require '[clojure.string :as str])
+  (->> test-data
+       (filter #(contains? (:attrs %) "JAR_PACKAGES"))
+       (map #(into
+               {:jar-packages (str/split (get-in % [:attrs "JAR_PACKAGES"]) #",")
+                :pom-parent   (get-in % [:attrs "POM_PARENT"])}
+               (dissoc % :attrs)
+               )
+            )
+       )
+  )
+
+(defn extract-packages
+  "Extract packages from artifact map as sequence of vectors."
+  [artifact-map]
+  (as-> (get-in artifact-map [:jar-packages]) results
+        (map #(vector % (dissoc artifact-map :jar-packages)) results)
+        )
+  )
+
+(defn reduce-packages
+  "Combines like artifacts by package"
+  [package-map next-package]
+  (if (not (contains? package-map (first next-package)))
+        (assoc package-map (first next-package) (list (last next-package)))
+        (update package-map (first next-package) #(cons (last next-package) %))
+        )
+  )
+
+(defn package-search
+  "Search for packages"
+  [package-name hierarchy]
+  (as-> (filter #(contains? (:attrs %) "JAR_PACKAGES") hierarchy) results
+        (map #(into
+                {:jar-packages (str/split (get-in % [:attrs "JAR_PACKAGES"]) #",")
+                 :pom-parent   (get-in % [:attrs "POM_PARENT"])}
+                (dissoc % :attrs))
+             results)
+        (mapcat extract-packages results)
+        (reduce reduce-packages {} results)
+        (get results package-name)
+        )
+  )
+
+(comment
+  (as-> (gather-hierarchy "ddf" "2.19.5") results
+        (filter #(contains? (:attrs %) "JAR_PACKAGES") results)
+        (map #(into
+                {:jar-packages (str/split (get-in % [:attrs "JAR_PACKAGES"])
+                                          #",")
+                 :pom-parent   (get-in % [:attrs "POM_PARENT"])}
+                (dissoc % :attrs))
+             results)
+        (mapcat extract-packages results)
+        (reduce reduce-packages {} results)
+        )
+  (package-search "ddf.catalog.operation.impl" (gather-hierarchy "ddf" "2.19.5"))
```

Comment:
Oh I see you're filtering the hierarchy for packages. What we actually want to do is make this a function of a single argument. Then expose a new function on the Java side that plugs the package string into a proper criteria query. RIght now we're not leveraging the indexed packages field. 

We can add a new Clojure wrapper function for the new Java function that calls `artifact-info->map` appropriately. All your existing code will work since the sequence of results are still artifact info maps, it's just the semantics for the search will be correct now.